aces/Loris
# [Test] Adding test coverage metric tool

This PR makes changes to the PHPUnit configuration in order to generate coverage metrics. The reporting can be viewed on Travis. There is also a tool that can be run locally, which generates detailed HTML files under the `htdocs` directory. These files can then be viewed on your browser.
 
#### Testing instructions
To see the metrics on Travis:
1. Checkout the **unit test** build on Travis. It should display some coverage metrics at the bottom.
To see the detailed HTML reports:
1. Checkout this branch
2. Run the tool `code_coverage.php` in the tools directory with: `php code_coverage.php`
3. See that the HTML files are properly created under `htdocs/log/codeCoverage/<date>`.
4. To view overall metrics and other useful information, go to `https://<loris-url>/log/codeCoverage/<date>/index.html` and `https://<loris-url>/log/codeCoverage/<date>/dashboard.html`
5. To view the code coverage for a specific library (ex. Utility.class.inc), go to `https://<loris-url>/log/codeCoverage/<date>/Utility.class.inc`

#### Proposal
The current proposal is that whenever unit tests are run (locally or on Travis), a command-line version of the metrics will be displayed for the PHP libraries. The tool used to generate the detailed HTML files can be run locally on any branch from the `tools` directory. 

#### TODO (If this correctly passes Travis)

- [x] Add date/time that coverage was calculated at the top of each HTML file. Keep track of past HTML files. 

- [x] Configure files to highlight most important points.
